src/win/util.c

```diff
@@ -484,7 +484,7 @@ uint64_t uv__hrtime(double scale) {
    * performance counter interval, integer math could cause this computation
    * to overflow. Therefore we resort to floating point math.
    */
-  return (uint64_t) ((double) counter.QuadPart * hrtime_interval_ * scale);
+  return (uint64_t) (counter.QuadPart * (uint64_t) scale / hrtime_interval_);
```

Comment:
And to reiterate, the correct solution for timing tests is to ignore process.hrtime() completely, and instead have an API that works with QPC values directly to calculate deltas (see my HRTimer example above).

This is on the MSDN magazine page for HRTimers. Note the sentence at the end.

>- Integer division will lose the remainder. This can cause loss of precision in some cases.
>- Conversion between 64 bit integers and floating point (double) can cause loss of precision because the floating point mantissa can't represent all possible integral values.
>- Multiplication of 64 bit integers can result in integer overflow.
>
>As a general principle, delay these computations and conversions as long as possible to avoid compounding the errors introduced.